### PR TITLE
MySQL support for uploads

### LINKED ISSUE #29675: Create tables from CSV file uploads

### PR DESCRIPTION
**Pull request by calherries**
Continuing on from #29922 , this resolves https://github.com/metabase/metabase/issues/29675 and https://github.com/metabase/metabase/issues/29676 for MySQL.

It implements the multimethod `driver/upload-type->database-type` for MySQL.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29829)

### CODE REVIEW

File: src/metabase/csv.clj [hunk 1]
```diff
@@ -108,10 +108,13 @@
...
+  "Loads a table from a CSV file. Returns the name of the newly created table.
+   `schema-name` can be nil for databases that don't support schemas."
   [driver database schema-name ^File file]
-  (let [table-name (file->table-name file)]
-    (driver/load-from-csv driver database schema-name table-name file)
+  (let [table-name        (file->table-name file)
+        schema+table-name (cond->> table-name
+                            (some? schema-name)
+                            (str schema-name "."))]
+    (driver/load-from-csv driver database schema+table-name file)
```
Comment (calherries) [inline src/metabase/csv.clj:119]: Note we now call `driver/load-from-csv` with `schema-name` and `table-name` combined already.

**Comment by tsmacdonald** [comment]
*[`src/metabase/driver/mysql.clj` line 641 at r2](https://reviewable.io/reviews/metabase/metabase/29829#-NSKvprV8MHl4nwzyND7:-NSKvprV8MHl4nwzyND8:b-oh0db2) ([raw file](https://github.com/metabase/metabase/blob/28b42535486939177b1994a4bb1727ffb76010c5/src/metabase/driver/mysql.clj#L641)):*
> ```Clojure
>    ::csv/text        identity
>    ::csv/int         #(Integer/parseInt %)
>    ::csv/float       #(Float/parseFloat %)
> ```

Should we use `DOUBLE` and `BIGINT` instead of `FLOAT` and `INTEGER` (and then `parse-double` and `parse-long` instead of `#(Integer/...)` and `#(Float/...)`)?

This probably applies to the Postgres one too, I just didn't think of it then.

You probably have a much better sense of what type conventions are for MySQL folk.

**Comment by tsmacdonald** [comment]
*[`src/metabase/driver/mysql.clj` line 642 at r2](https://reviewable.io/reviews/metabase/metabase/29829#-NSKwsFk1vEi7a7UGtOa:-NSKwsFk1vEi7a7UGtOb:bpslmy7) ([raw file](https://github.com/metabase/metabase/blob/28b42535486939177b1994a4bb1727ffb76010c5/src/metabase/driver/mysql.clj#L642)):*
> ```Clojure
>    ::csv/int         #(Integer/parseInt %)
>    ::csv/float       #(Float/parseFloat %)
>    ::csv/boolean     read-string})
> ```

unfortunately this will need its own parser: we support all of

```
#"(?i)true|t|yes|y|1|false|f|no|n|0"
```

So something like

```
(defn parse-boolean
  [s]
  (cond
     (re-matches #"(?i)true|t|yes|y|1" (str/trim s)) true
     (re-matches #"(?i)false|f|no|n|0" (str/trim s)) false
     :else (throw ...)))
```

File: test/metabase/api/card_test.clj [hunk 2]
```diff
@@ -2545,21 +2545,65 @@
...
   (mt/test-driver :postgres
     (mt/with-empty-db
-      (let [db-id              (u/the-id (mt/db))
-            existing-table-ids (t2/select-pks-vec Table :db_id db-id)
-            file               (upload-test/csv-file-with
-                                ["id, name"
-                                 "1, Luke Skywalker"
-                                 "2, Darth Vader"]
-                                "filename")
-            upload!            (fn []
```
Comment (tsmacdonald) [inline test/metabase/api/card_test.clj:2559]: definitely nicer to have this as a standalone fn, thanks

**Review by tsmacdonald — APPROVED** [review]

**Pull request merged**